src/table/settings.ini

```diff
@@ -6,47 +6,49 @@
-static bool InvalidateCompanyWindow(int32 p1);
-static bool ZoomMinMaxChanged(int32 p1);
-static bool SpriteZoomMinChanged(int32 p1);
-static bool MaxVehiclesChanged(int32 p1);
-static bool InvalidateShipPathCache(int32 p1);
+static void v_PositionMainToolbar(int32 new_value);
+static void v_PositionStatusbar(int32 new_value);
+static void PopulationInLabelActive(int32 new_value) { UpdateAllTownVirtCoords(); }
+static void RedrawScreen(int32 new_value) { MarkWholeScreenDirty(); }
+static void RedrawScreen(const std::string &newval) { MarkWholeScreenDirty(); }
+static void RedrawSmallmap(int32 new_value);
+static void StationSpreadChanged(int32 new_value);
+static void InvalidateBuildIndustryWindow(int32 new_value) { InvalidateWindowData(WC_BUILD_INDUSTRY, 0); }
+static void CloseSignalGUI(int32 new_value);
+static void InvalidateTownViewWindow(int32 new_value) { InvalidateWindowClassesData(WC_TOWN_VIEW, new_value); }
+static void DeleteSelectStationWindow(int32 new_value) { DeleteWindowById(WC_SELECT_STATION, 0); }
+static void UpdateConsists(int32 new_value);
+static void TrainAccelerationModelChanged(int32 new_value);
+static void RoadVehAccelerationModelChanged(int32 new_value);
+static void TrainSlopeSteepnessChanged(int32 new_value);
+static void RoadVehSlopeSteepnessChanged(int32 new_value);
+static void DragSignalsDensityChanged(int32 new_value) { InvalidateWindowData(WC_BUILD_SIGNAL, 0); }
+static void TownFoundingChanged(int32 new_value);
+static void DifficultyNoiseChange(int32 new_value);
+static void MaxNoAIsChange(int32 new_value);
+static bool CheckRoadSide(int &new_value);
+static bool CheckMaxHeightLevel(int32 &new_value);
+static void UpdateMaxHeightLevel(int32 new_value) { InvalidateWindowClassesData(WC_SMALLMAP, 2); }
+static bool CheckFreeformEdges(int32 &new_value);
+static void UpdateFreeformEdges(int32 new_value);
+static bool CheckDynamicEngines(int32 &new_value);
+static void StationCatchmentChanged(int32 new_value);
+static void InvalidateVehTimetableWindow(int32 new_value) { InvalidateWindowClassesData(WC_VEHICLE_TIMETABLE, VIWD_MODIFY_ORDERS); }
+static void InvalidateCompanyLiveryWindow(int32 new_value);
+static void InvalidateNewGRFChangeWindows(int32 new_value);
+static void InvalidateIndustryViewWindow(int32 new_value) { InvalidateWindowClassesData(WC_INDUSTRY_VIEW); }
+static void InvalidateAISettingsWindow(int32 new_value) { InvalidateWindowClassesData(WC_AI_SETTINGS); }
+static void RedrawTownAuthority(int32 new_value) { SetWindowClassesDirty(WC_TOWN_AUTHORITY); }
+static void InvalidateCompanyInfrastructureWindow(int32 new_value) { InvalidateWindowClassesData(WC_COMPANY_INFRASTRUCTURE); }
+static void InvalidateCompanyWindow(int32 new_value) { InvalidateWindowClassesData(WC_COMPANY); }
+static void ZoomMinMaxChanged(int32 new_value);
+static void SpriteZoomMinChanged(int32 new_value);
+static void MaxVehiclesChanged(int32 new_value);
+static void InvalidateShipPathCache(int32 new_value);
```

Comment:
one thing I do not like one bit with this pre/post, is that it is very difficult to spot what functions are pre, and which are post. It is in the nuance:
For ints, it is a single `&`.
For strings, it is a `const` marker.
This is really hard to read. And the naming being all over the place doesn't help. Maybe if we already order it based on pre/post, that could help, but not sure.

Again not sure if that is something for this PR, but this PR does make it worse than it was :) Some kind of naming convention might seriously help here.

Just some examples of what frustrates:
`MaxNoAIsChange` vs `ZoomMinMaxChanged` vs `UpdateMaxHeightLevel`
`CanUpdateServiceInterval` vs `CheckDynamicEngines` vs `ReplaceAsteriskWithEmptyPassword`

It is often not even clear they are settings callbacks, and feel like anything can use them (which might even be true).

Reply:
The easiest way to distinguish between 'pre' and 'post' is by the return parameter. When it is a `bool`, then it is a pre check callback. When it is a `void` it is a post update callback. Previously you had no clue at all, so that's already definitely an improvement ;)

I agree the naming is inconsistent, though changing naming and/or moving things around makes reviewing more difficult.

Reply:
As said, doesn't have to be for this PR, but might be a good cleanup to do somewhere soon. Similar with the other related comments .. I also noticed I find it rather difficult to add stuff because of this.

That said, in general, the settings.ini is messy .. very hard to find anything, as it is just an endless list of shit .. but one thing at the time :)